float32 posX
float32 posY
float32 velX
float32 velY
